torchvision/ops/boxes.py

```diff
@@ -133,6 +134,54 @@ def clip_boxes_to_image(boxes: Tensor, size: Tuple[int, int]) -> Tensor:
     return clipped_boxes.reshape(boxes.shape)
 
 
+def box_convert(boxes: Tensor, in_fmt: str, out_fmt: str) -> Tensor:
+    """
+    Converts boxes from given in_fmt to out_fmt.
+
+    Arguments:
+        boxes (Tensor[N, 4]): boxes which will be converted.
+        in_fmt (str): Input format of given boxes.
+        out_fmt (str): Output format of given boxes.
+
+    Returns:
+        boxes (Tensor[N, 4]): Boxes into converted format.
+    """
+    allowed_fmts = ("xyxy", "xywh", "cxcywh")
+    assert in_fmt in allowed_fmts
+    assert out_fmt in allowed_fmts
+
+    if in_fmt == out_fmt:
+        # return boxes.clone()  # to ensure always returning a copy
+        boxes_converted = boxes.clone()
+        return boxes_converted
+
+    else:
+        if in_fmt != 'xyxy' and out_fmt != 'xyxy':
+            if in_fmt == "xywh":
+                boxes_xyxy = _box_xywh_to_xyxy(boxes)
+                if out_fmt == "cxcywh":
+                    boxes_converted = _box_xyxy_to_cxcywh(boxes_xyxy)
+
+            elif in_fmt == "cxcywh":
+                boxes_xyxy = _box_cxcywh_to_xyxy(boxes)
+                if out_fmt == "xywh":
+                    boxes_converted = _box_xyxy_to_xywh(boxes_xyxy)
```

Comment:
While this is fine, my first thought was to so something like the following
```python
            if in_fmt == "xywh":
                boxes = _box_xywh_to_xyxy(boxes)
                in_fmt = "xyxy"
            elif in_fmt == "cxcywh":
                boxes = _box_cxcywh_to_xyxy(boxes)
                in_fmt = "xyxy"
```
and let the rest of the dispatch to be done in the last branch. This way, we don't need to replicate the `out` dispatch logic here.

You don't need to change this here so that we can move forward quickly with this PR, but it would be good to send a follow-up PR improving this part after this PR gets merged. Thoughts?

Reply:
I guess I will leave this for next PR, my original idea was we support direct conversions at some point of time and it should be simpler to refactor for future. But this too works fine.